lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -274,6 +275,15 @@ public function getSchemaFromMetadata(array $classes)
 
             if (isset($class->table['uniqueConstraints'])) {
                 foreach ($class->table['uniqueConstraints'] as $indexName => $indexData) {
+                    $uniqIndex = new Index($indexName, $indexData['columns'], true, false, [], isset($indexData['options']) ? $indexData['options'] : []);
+
+                    foreach ($table->getIndexes() as $tableIndexName => $tableIndex) {
+                        if ($tableIndex->isFullfilledBy($uniqIndex)) {
+                            $table->dropIndex($tableIndexName);
+                            break;
```

Comment:
I assumed if one fulfilled index will be found it has no sense to look for another one. Can be removed, but is it possible to have more than one fulfilled index in table before? Can you provide an example? 

I was thinking to add condition to check if found index is primary, than we shouldn't add unique index, but it will be only possible when annotation is invalid - so when we have `uniqueConstraint` at primary column.